tests/drivers/disk/disk_access/src/main.c

```diff
@@ -14,18 +14,30 @@
 #include <zephyr/storage/disk_access.h>
 #include <zephyr/device.h>
 
+#ifdef CONFIG_DISK_DRIVER_LOOPBACK
+#include <ff.h>
+#include <zephyr/fs/fs.h>
+#include <zephyr/drivers/loopback_disk.h>
+#endif
+
 #if defined(CONFIG_DISK_DRIVER_SDMMC)
-#define DISK_NAME CONFIG_SDMMC_VOLUME_NAME
+#define DISK_NAME_PHYS CONFIG_SDMMC_VOLUME_NAME
 #elif IS_ENABLED(CONFIG_DISK_DRIVER_MMC)
-#define DISK_NAME CONFIG_MMC_VOLUME_NAME
+#define DISK_NAME_PHYS CONFIG_MMC_VOLUME_NAME
 #elif IS_ENABLED(CONFIG_DISK_DRIVER_RAM)
-#define DISK_NAME "RAM"
+#define DISK_NAME_PHYS "RAM"
 #elif IS_ENABLED(CONFIG_NVME)
-#define DISK_NAME "nvme0n0"
+#define DISK_NAME_PHYS "nvme0n0"
 #else
 #error "No disk device defined, is your board supported?"
 #endif
 
+#ifdef CONFIG_DISK_DRIVER_LOOPBACK
+#define DISK_NAME "loopback0"
+#else
+#define DISK_NAME DISK_NAME_PHYS
+#endif
```

Comment:
Why not just add this to the above set of #if #else.
Does the FAT FS mounting work? The Fat FS has very limited number of mount points it can work with and these mountpoints are used also as disk names.
https://github.com/zephyrproject-rtos/zephyr/blob/51d80a9838b3052f9a564d3a8df689b9501597ed/modules/fatfs/zephyr_fatfs_config.h#L84-L95
https://github.com/zephyrproject-rtos/fatfs/blob/427159bf95ea49b7680facffaa29ad506b42709b/include/ffconf.h#L174C9-L174C23

Reply:
> Why not just add this to the above set of #if #else.

What exactly do you mean? In the loopback case, I need both `DISK_NAME` (for the actual disk access tests) and `DISK_NAME_PHYS` (for the setup of the loopback disk access).

> Does the FAT FS mounting work?

The tests pass, so I presume it does? I believe it's using the RAM disk.

Reply:
Ah right. The FAT is not mounted on the loopback drive, the fat is the source of loopback image only